### PR TITLE
Add missing foreign keys to Organization and User tables (Phase 2)

### PR DESCRIPTION
**Pull request by yoomlam**
Bumps #14732
* This is phase 2 of a [multi-phase plan](https://hackmd.io/wum2LtUfSHC-1o6-peBseQ) to add missing foreign keys.
* Phase 1 #16034 has more details.

### Description
Add foreign keys to the `organizations` and `users` table. 

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
Before deployment, rerun the following to ensure the foreign keys will be valid:
```ruby
IhpDraft.includes(:organization).where.not(organization_id: nil).select{|r| r.organization == nil}.count
=>0

ou=OrganizationsUser.includes(:organization).where.not(organization_id: nil).select{|r| r.organization == nil};
ou.count
=>45
org_ids = ou.map(&:organization_id).uniq
=> [130, 139, 408, 163, 171, 190, 412, 320]
=> [130, 139, 408, 163, 171, 183, 190, 212, 412, 320] # Apr 12th

org_ids.count
=> 10
# Check inactive orgs
ou2=Organization.unscoped.find(org_ids).count
=> 10
# All 10 exists

OrganizationsUser.includes(:user).where.not(user_id: nil).select{|r| r.user == nil}.count
=> 0

vc=VsoConfig.includes(:organization).where.not(organization_id: nil).select{|r| r.organization == nil};
vc.count
=> 0

JobNote.includes(:user).where.not(user_id: nil).select{|r| r.user == nil}.count
=>0
Message.includes(:user).where.not(user_id: nil).select{|r| r.user == nil}.count
=>0
```

Also check and fix records in UAT as well.

### Database Changes

* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: spec/controllers/organizations/users_controller_spec.rb [hunk 1]
```diff
@@ -268,24 +268,37 @@
     end
   end
 
-  describe "DELETE /organizations/:org_url/users/:user_id", skip: "Flake" do
+  describe "DELETE /organizations/:org_url/users/:user_id" do
```
Comment (yoomlam) [inline spec/controllers/organizations/users_controller_spec.rb:271]: Moved `, skip: "Flake"` to the sub`context` so I can add a new unskipped context.

File: spec/controllers/organizations/users_controller_spec.rb [hunk 2]
```diff
@@ -268,24 +268,37 @@
...
+
+    before { User.stub = bva_admin_user }
+
+    context "when user is a non-judge in the organization" do
+      it "removes user from organization" do
+        expect(OrganizationsUser.where(organization: org).count).to eq(2)
+        subject
+        resp = JSON.parse(response.body)
+
+        expect(OrganizationsUser.where(organization: org).count).to eq(1)
```
Comment (yoomlam) [inline spec/controllers/organizations/users_controller_spec.rb:295]: Check that `has_many :organizations_users, dependent: :destroy` in `Organization` works.

File: spec/models/organizations_user_spec.rb [hunk 3]
```diff
@@ -71,6 +71,25 @@
...
+      expect(organization.users.count).to eq(1)
+      expect(OrganizationsUser.count).to eq(1)
+      subject
+      # User is no longer part of the inactive organization
+      expect(user.organizations).to_not include organization
+
+      # However, associations queried from the organization are unchanged
+      expect(organization.users.count).to eq(1)
+      # because record in OrganizationsUser table still exists
+      expect(OrganizationsUser.count).to eq(1)
```
Comment (yoomlam) [inline spec/models/organizations_user_spec.rb:89]: `default_scope` prevents an inactive Organization from being queried without using `Organization.unscoped`.

https://github.com/department-of-veterans-affairs/caseflow/blob/40f3f77ecf7229dce71c7aa89256d199f37eddf0/app/models/organization.rb

**Comment by yoomlam** [comment]
Data in Prod and UAT DBs satisfy the the foreign keys constraints in this PR.

**Review by nanotone — APPROVED** [review]
:shipit:

**Pull request merged**